runtime/compiler/p/codegen/CallSnippet.cpp

```diff
@@ -597,7 +593,24 @@ uint8_t *TR::PPCVirtualUnresolvedSnippet::emitSnippetBody()
     * This is used for private nestmate calls.
     */
    *(intptrj_t*)cursor = (intptrj_t)thunk;
-   //TODO: Handle relocation here for the thunk. Currently not supported by AOT.
+
+   auto info =
+      (TR_RelocationRecordInformation *)comp->trMemory()->allocateMemory(
+         sizeof (TR_RelocationRecordInformation),
+         heapAlloc);
+
+   // data1 = constantPool
+   info->data1 = cpAddr;
+
+   // data2 = inlined site index
+   info->data2 = callNode ? callNode->getInlinedSiteIndex() : (uintptr_t)-1;
+
+   // data3 = distance in bytes from Constant Pool Pointer to J2I Thunk
+   info->data3 = 3 * TR::Compiler->om.sizeofReferenceAddress();
```

Comment:
Personally, I'd be happier if this offset was computed from the generated code addresses...something like `(cursorForThunkAddress - cursorForConstantPoolPointer)` so that, even if the layout changes, nobody has to remember to change this `3` or the other `8`. Yeah, I know that means you'll have to "remember" more locations so that you can generate the relocation later on, but I think it's a better approach.

Reply:
At that point in the code cursor and j2iThunkRelocationPoint already have the values I need so I replaced it with cursor - j2iThunkRelocationPoint.